common/persistence/nosql/nosqlplugin/cassandra/tasks.go

```diff
@@ -89,10 +89,56 @@ func toTaskListPartitionConfig(v interface{}) *persistence.TaskListPartitionConf
 	version := partition["version"].(int64)
 	numRead := partition["num_read_partitions"].(int)
 	numWrite := partition["num_write_partitions"].(int)
+	readPartitions := toTaskListPartitions(partition["read_partitions"])
+	writePartitions := toTaskListPartitions(partition["write_partitions"])
+	// If they're out of sync, go with the value of num_*_partitions. This is necessary only while support for
+	// read_partitions and write_partitions rolls out
```

Comment:
until `read_partitions`/`write_partitions` are written back to DB we will be resetting the partition mappings. I guess it doesn't matter for now as isolation group feature is disabled